src/contracts/GPv2Settlement.sol

```diff
@@ -100,24 +106,128 @@ contract GPv2Settlement {
+        uint256[] calldata clearingPrices,
+        bytes calldata encodedTrades
+    )
+        internal
+        view
+        returns (
+            GPv2AllowanceManager.Transfer[] memory inTransfers,
+            GPv2AllowanceManager.Transfer[] memory outTransfers
+        )
+    {
+        uint256 tradeCount = encodedTrades.tradeCount();
+        inTransfers = new GPv2AllowanceManager.Transfer[](tradeCount);
+        outTransfers = new GPv2AllowanceManager.Transfer[](tradeCount);
+
+        GPv2Encoding.Trade memory trade;
+        for (uint256 i = 0; i < tradeCount; i++) {
+            encodedTrades.tradeAtIndex(i).decodeTrade(
+                domainSeparator,
+                tokens,
+                trade
+            );
+            processTrade(
+                trade,
+                clearingPrices[trade.sellTokenIndex],
+                clearingPrices[trade.buyTokenIndex],
+                inTransfers[i],
+                outTransfers[i]
+            );
+        }
+    }
+
+    /// @dev Process trades for a single EOA order.
+    /// @param trade The trade to process.
+    /// @param sellPrice The price of the order's sell token.
+    /// @param buyPrice The price of the order's buy token.
+    /// @param inTransfer Memory location to set computed tranfer into the
+    /// settlement contract to execute trade.
+    /// @param outTransfer Memory location to set computed transfer out to order
+    /// owner to execute trade.
+    function processTrade(
+        GPv2Encoding.Trade memory trade,
+        uint256 sellPrice,
+        uint256 buyPrice,
+        GPv2AllowanceManager.Transfer memory inTransfer,
+        GPv2AllowanceManager.Transfer memory outTransfer
+    ) internal pure {
+        GPv2Encoding.Order memory order = trade.order;
+        // NOTE: Currently, the above instanciation allocates an unitialized
+        // `Order` that gets never used. Adjust the free memory pointer to free
+        // the unused memory by subtracting `sizeof(Order) == 288` bytes.
+        // <https://solidity.readthedocs.io/en/v0.7.5/internals/layout_in_memory.html>
+        // TODO: Remove this once the following fix is merged and released:
+        // <https://github.com/ethereum/solidity/pull/10341>
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            mstore(0x40, sub(mload(0x40), 288))
+        }
+
+        inTransfer.owner = trade.owner;
+        inTransfer.token = order.sellToken;
+        outTransfer.owner = trade.owner;
+        outTransfer.token = order.buyToken;
+
+        if (order.kind == GPv2Encoding.OrderKind.Sell) {
+            uint256 executedSellAmount;
+            if (order.partiallyFillable) {
+                executedSellAmount = trade.executedAmount;
+            } else {
+                executedSellAmount = order.sellAmount;
+            }
+
+            inTransfer.amount = executedSellAmount;
+            // NOTE: Don't use `SafeMath.div` here as it allocates a string even
```

Comment:
Is that also true for the other operations? Maybe something to raise to the open zeppelin team?

Reply:
> Is that also true for the other operations?

No. Its because just for division, they have two overrides:
```
function div(uint256, uint256)
function div(uint256, uint256, string memory reason)
```

And the former calls the latter which causes the compiler to allocate a `string memory` for the call.

> Maybe something to raise to the open zeppelin team?

Good idea. I had planned on doing it, just not sure the best way to solve the issue.